display_list/display_list_image_filter.h

```diff
@@ -723,6 +727,10 @@ class DlLocalMatrixImageFilter final : public DlImageFilter {
     return skia_object->makeWithLocalMatrix(matrix_);
   }
 
+  MatrixCapability matrix_capability() const override {
+    return MatrixCapability::kComplex;
```

Comment:
Shouldn't this defer to the wrapped filter? Otherwise you could wrap a weak filter with a translate and then wrap again with anything you want.

Reply:
Done

Interestingly Skia's equivalent class is always returning `kComplex`:
https://skia.googlesource.com/skia/+/refs/heads/main/src/core/SkLocalMatrixImageFilter.h#30

Possibly a bug?

Reply:
The Skia code doesn't call that virtual method directly -- the non-virtual base class method handles taking the minimum compatability against a filter and all of its children: https://source.chromium.org/chromium/chromium/src/+/main:third_party/skia/src/core/SkImageFilter.cpp;l=327;drc=44798fcb6d921c4a8fee8331289021a3b25b4450